### PR TITLE
kubespy: init at 0.5.1

### LINKED ISSUE #92196: add kubespy
**Project description**
_A tool to observe Kubernetes resources in real time_

**Metadata**

* homepage URL: https://github.com/pulumi/kubespy
* source URL: https://github.com/pulumi/kubespy
* license: asl20
* platforms: unix

### PR DESCRIPTION
**Pull request by blaggacao**
###### Motivation for this change

A tool to observe Kubernetes resource in real time
closes #92196

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by blaggacao** [comment]
figuring out build issue upstream: https://github.com/pulumi/kubespy/issues/73

**Comment by ludovicc** [comment]
##### Reviewed points

- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package build on Darwin
- [x] executables tested on Darwin
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.platforms` is set
- [x] `meta.maintainers` is set
- [x] build time only dependencies are declared in `nativeBuildInputs`
- [x] source is fetched using the appropriate function
- [x] phases are respected
- [x] patches that are remotely available are fetched with `fetchpatch`

##### Possible improvements

##### Comments

Tested only execution of kubespy with no arguments, it prints help for the command.

**Comment by blaggacao** [comment]
@ludovicc thank you anlot for this review on my first PR to nicpkgs.

/status needs_merger

File: pkgs/applications/networking/cluster/kubespy/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{ lib, buildGoModule, fetchFromGitHub, installShellFiles }:
+
+buildGoModule rec {
+  pname = "kubespy";
+  version = "0.5.1";
+
+  goPackagePath = "github.com/pulumi/kubespy";
```
Comment (jonringer) [inline pkgs/applications/networking/cluster/kubespy/default.nix:7]: i believe this is unused with buildGoModule
```suggestion
```

**Review by jonringer — COMMENTED** [review]
otherwise LGTM

File: pkgs/applications/networking/cluster/kubespy/default.nix (hunk 1 above)
Comment (blaggacao) [reply pkgs/applications/networking/cluster/kubespy/default.nix]: Thanks! Addressed and force-pushed.

**Review by jonringer — APPROVED** [review]
diff LGTM
commits LGTM

```
https://github.com/NixOS/nixpkgs/pull/92197
1 package built:
kubespy
```

**Comment by Luflosi** [comment]
Minor nitpick: this PR added two extra spaces in an otherwise empty line in `pkgs/top-level/all-packages.nix`. This isn't the case anywhere else in this file.

**Comment by blaggacao** [comment]
@Luflosi Thanks for hinting, looks like I might need to tweak my nix-lsp.

**Pull request merged**